DTAConfig/CustomSettings/FileSourceDestinationInfo.cs

```diff
@@ -0,0 +1,115 @@
+﻿using ClientCore;
+using Rampastring.Tools;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DTAConfig.CustomSettings
+{
+    sealed class FileSourceDestinationInfo
+    {
+        public FileSourceDestinationInfo(string source, string destination, FileOperationOptions options)
+        {
+            SourcePath = source;
+            DestinationPath = destination;
+            FileOperationOptions = options;
+        }
+
+        public string SourcePath { get; }
+        public string DestinationPath { get; }
+        public string CachedPath => ProgramConstants.ClientUserFilesPath + "SettingsCache/" + SourcePath;
+        public FileOperationOptions FileOperationOptions { get; }
+
+        public void Apply()
+        {
+            switch (FileOperationOptions)
+            {
+                case FileOperationOptions.OverwriteOnMismatch:
+                    string sourceHash = Utilities.CalculateSHA1ForFile(ProgramConstants.GamePath + SourcePath);
+                    string destinationHash = Utilities.CalculateSHA1ForFile(ProgramConstants.GamePath + DestinationPath);
+
+                    if (sourceHash != destinationHash)
+                    {
+                        File.Copy(ProgramConstants.GamePath + SourcePath,
+                            ProgramConstants.GamePath + DestinationPath, true);
+                    }
+
+                    break;
+
+                case FileOperationOptions.DontOverwrite:
+                    if (File.Exists(ProgramConstants.GamePath + DestinationPath))
+                        break;
+
+                    File.Copy(ProgramConstants.GamePath + SourcePath,
+                        ProgramConstants.GamePath + DestinationPath, false);
```

Comment:
Looking at the examples above and below, it'd be better to revert the `if` condition here and have the operation in brackets, keeping the code consistent and leaving only one `break` statement and exit point from this case.